inc/class-layouts.php

```diff
@@ -396,4 +401,65 @@ function content_loop( $post_id, $i, $args = array() ) {
 		return apply_filters( 'sell_media_filter_content_loop', $html, $original_id, $i );
 	}
 
+	// Add image license metadata on archive page
+	function sell_media_add_license_meta_loop($html, $post_id, $i) {
+
+		global $mime_type;
+		
+		if ( post_password_required( $post_id ) && sell_media_is_search() ) {
+			return;
+		}
+		if ( 'attachment' === get_post_type( $post_id ) ) {
+			$attachment_id = $post_id; // always and attachment
+			$post_id = get_post_meta( $attachment_id, $key = '_sell_media_for_sale_product_id', true ); // always a sell_media_item
+		} else {
+			$attachment_id = sell_media_get_attachment_id( $post_id ); // always an attachment
+		}
+		if (!$attachment_id) {
+			return $html;
+		}
+		$_output = '<script type="application/ld+json">'.$this->sell_media_get_all_sizes_images($attachment_id).'</script>';
+		
+		return $html . $_output;
+	}
+
+	// Add image license metadata in media gallery page, single page and collection page
+	function sell_media_add_license_meta_gallery($html, $post_id) {
+
+		$_output = '<script type="application/ld+json">';
+		if (sell_media_has_multiple_attachments( $post_id )) {
+			$attachment_ids = sell_media_get_attachments( $post_id );	
+			if ( $attachment_ids ) foreach ( $attachment_ids as $attachment_id ) {
+				$_output .= $this->sell_media_get_all_sizes_images($attachment_id);
+			}
+		} else {
+			$_output .= $this->sell_media_get_all_sizes_images($post_id);
+		}
+		$_output .= '</script>';
+		return $html . $_output;
+	}
+
+	/**
+	 * Get all images sizes from attachment id
+	 *
+	 * @param int $attachment_id
+	 * @return json licence content
+	 */
+	function sell_media_get_all_sizes_images( $attachment_id ) {
```

Comment:
@ashok-ladumor We only need to make the image size that Google indexes in search as licensable. We don't need to loop over all sizes of an image. They are all the same image, just different crops.